### PR TITLE
Feature/add log exception with id for android

### PR DESCRIPTION
**Pull request by easybird**
add an extra implementation for Android to improve fabric statistics

### CODE REVIEW

File: android/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.gradle
```
Comment (eliaslecomte) [inline android/.gitignore]: 🌮 👍

File: android/src/main/java/com/smixx/fabric/SMXCrashlytics.java [hunk 2]
```diff
@@ -33,6 +37,36 @@ public void logException(String value) {
...
+    @ReactMethod
+    public void logExceptionWithId(String id, String error) {
+        logExceptionWithId(id, error, "");
+    }
+
+    @ReactMethod
+    public void logExceptionWithId(String id, String error, String source) {
+        Exception exception = new Exception(id);
+
+        List<StackTraceElement> arrayList = new ArrayList<StackTraceElement>();
```
Comment (eliaslecomte) [inline android/src/main/java/com/smixx/fabric/SMXCrashlytics.java]: You can replace this method with:
final Exception exception = new Exception(id);
final StackTraceElement stackTraceElement = new StackTraceElement(id, source != null ? source : "", 0);
final StackTraceElement[] stackTraceArray = new StackTraceElement[1];
stackTraceArray[0] = stackTraceElement;
exception.setStackTrace(stackTraceArray);
Crashlytics.logException(exception);

Comment (easybird) [reply android/src/main/java/com/smixx/fabric/SMXCrashlytics.java]: lol funny that you say that as we wrote the code together 🙈

Comment (bitcrumb) [reply android/src/main/java/com/smixx/fabric/SMXCrashlytics.java]: New pull request! 🤓

Comment (eliaslecomte) [reply android/src/main/java/com/smixx/fabric/SMXCrashlytics.java]: I know, woops.

**Comment by eliaslecomte** [comment]
Why only on Android?

**Comment by easybird** [comment]
@eliaslecomte because we only need this to improve android fabric analytics. The stacktraces from ios get "parsed" better in the fabric dashboard (as seperate error).
Why did we not implement anything for ios? The original lib already implemented one method for android and one for ios 🙈 I know, very strange for a bridge...

**Pull request merged**